third_party/move/move-model/src/model.rs

```diff
@@ -1303,10 +1304,8 @@ impl GlobalEnv {
         }
 
         let used_modules = self.get_used_modules_from_bytecode(&module);
-        let friend_modules = self.get_friend_modules_from_bytecode(&module);
```

Comment:
I think we need to keep this for the case we work from a CompiledModule. In general, we need to be able to construct a model from bytecode only, without access to the expansion AST, and then this is the only info we have. If this conflicts with your FriendDecl, you could have two functions like `get_friends_via_bytecode`  which accesses this field and your new function.